### ISSUE TITLE
Introducing weather Data

### ISSUE THREAD

**Issue by qaisaloweiti** [queraltab/Greenhouses-Library#18]
I've an inquiry regarding the weather data. How can we use them in the simulation?
I made some corrections to the model to match Open Modelica version 4, but I still have some errors regarding screen control "Control SC". Could you please help me?
Qais

**Comment by vekadvies** [comment]
If you fork this repo and push your corrections we can have a look. I’m very interested in getting this model running in OpenModelica.

**Comment by qaisaloweiti** [comment]
Yes, I can show the corrections made, but can you help me introduce the data (TMY) to the code ?
Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by vekadvies** [comment]
I'm not in any way experienced with this library, but had a look a while ago.
Data is loaded into the model by combitimetables reading data from text files.

These text files are in /Greenhouses/Resources/Data/
10Dec-22Nov.txt       - file with the climate data parameters.  You can replace this with you own data.
SP_10Dec-22Nov.txt  - file with temperature and co2 setpoints  (for each hour of the climate data)
SC_usable_10Dec-22Nov.txt - file with screen control setpoints (for each half hour, double number of climate data)

It looks like no screen controller is implemented in the model and actual screen positions have to be given to the model via the combitimetable.  This file you have to create based on you own climate data or implement a controller.

Does this help you in any way?

**Comment by qaisaloweiti** [comment]
I was still unable to import weather data. Can we have a video call via teams tomorrow (20/01/2022) at 10:30 a.m. (Amsterdam local time)?. If not, please recommend some people who are knowledgeable about thermal system engineering simulation using Modelica to contact with.

Thank you very much
Regards,
Qais Al-Oweiti



________________________________

**Comment by vekadvies** [comment]
Sorry can't help you there.

Please look at the documentation:
https://greenhouses-library.readthedocs.io/en/latest/examples.html

_The simulated greenhouse is located in Belgium and the simulation period is from the 10th of December to the 22nd of November. Two input data files are required:

Weather data: The input weather data for the simulation period is extracted from a TMY for Brussels and can be found in ‘Greenhouses/Resources/Data/10Dec-22Nov.txt’. The file contains data for the outside air temperature, air pressure, wind speed and global irradiation. The sky temperature, previously computed in a Python script, is also included in this file.
Climate control set-points: The temperature and CO2 set-points for the simulation period are calculated according to the strategy presented in [Control Systems](https://greenhouses-library.readthedocs.io/en/latest/control.html#control) and can be found in ‘Greenhouses/Resources/Data/SP_10Dec-22Nov.txt’.
These .txt files are accessed by means of TMY_and_control and SP_new, which are two CombiTimeTables models from the Modelica Standard Library._

And for the screen:
https://greenhouses-library.readthedocs.io/en/latest/control.html

_However, since this work does not focus on climate set-points optimization, no innovative control is proposed. Instead, the strategy proposed in [[1]](https://greenhouses-library.readthedocs.io/en/latest/zbibliography.html#aaslyng-intelligrow-2003) is implemented in Python and the set-points are inputted as a time-series “.txt” file in the model. This strategy consists in minimizing energy consumption while maintaining a crop growth close to the maximal growth rate_

**Comment by qaisaloweiti** [comment]
Thank you very much.

The weather data is clear to me and when I get the TMY of the desired region, I will be able to have the results for this location. I have an inquiry regarding
the equations used in model. Where can I find them? is there any reference for different correlations used?
For example in "solar model", could you advise me where can I find the reference of the function used to estimate the NIR transmission between the canopy and the floor tau_CF "Transmission and reflection between two layers" ?

I tried to do so in google but I did not find.

Thank you
Regards,
Qais
________________________________

**Comment by vekadvies** [comment]
From the documentation:
https://greenhouses-library.readthedocs.io/en/latest/greenhouse.html#solar-model

quote:
_. The NIR absorbed by the canopy and the floor is thus determined by considering the lumped cover, the canopy and the floor as a multiple layer model [[34]]  <--- reference 34!

https://greenhouses-library.readthedocs.io/en/latest/zbibliography.html

[34] | B. H. E. Vanthoor, C. Stanghellini, E. J. van Henten, and P. H. B. de Visser. A methodology for model-based greenhouse design: Part 1, a greenhouse climate model for a broad range of designs and climates. Biosystems Engineering, 110(4):363–377, December 2011. doi:10.1016/j.biosystemseng.2011.06.001.

[link](https://www.sciencedirect.com/science/article/abs/pii/S1537511011000948)
The greenhouse climate model used in this modelica model is based on this paper.

**Comment by qaisaloweiti** [comment]
I have a question regarding ventilation in order to ensure fresh air intake from the surroundings while maintaining temperature and relative humidity below a certain limit. In the control equation below, the output variable 'y' is determined by the values of CS from the temperature controller (PIDT.CS) and CS from the relative humidity controller (PID.CS).

y = 1/(1+exp(-200*(Mdot-0.05)))*max(PID.CS,PIDT.CS)

For a specific case study involving a greenhouse with a floor area of 40 square meters and a height of 3.5 meters, I would like to determine the appropriate value for 'y'.


In my proposed Modelica model, I have not considered the presence of a thermal screen. The heat accumulates throughout the day, resulting in a maximum air temperature inside the greenhouse reaching around 66 °C. This particular scenario took place on June 1st during summer, with the maximum outside temperature recorded as Tout = 40.5 °C.

It appears that the ventilation system is not performing optimally. However, when I adjust the value of U_vents to 0.3 or 0.4, the internal air temperature in the greenhouse decreases. On the other hand, when using u_vents_RH_T_Mdot, the ventilation system does not work as effectively.

In addition, I am currently working on incorporating an Effectiveness NTU heat exchanger from the buildings library. The aim is to effectively cool down the internal space, removing excess heat, while also maintaining humidity levels within acceptable limits during extreme winter and summer days. If you have any suggestions or recommendations, I would greatly appreciate your input.

I hope my issues are clear to you and hope to hear from you.
Thank you very much and regards,
Qais Aloweiti

[cid:494a41f2-8c95-43ce-ab61-74ac990492d4]
[cid:2b81d0b5-9988-442b-8382-9802ee84213c]
________________________________
